src/libdistgen.cpp

```diff
@@ -175,7 +191,34 @@ double distgend_is_membound_scaled(distgend_configT config) {
 static void *thread_benchmark(void *arg) {
 	thread_argsT *thread_args = (thread_argsT *)arg;
 	double *ret = (double *)calloc(1, sizeof(double));
+#ifdef __QNXNTO__
+	// QNX does not support pthread thread affinity. Set affinity here.
+	// TODO duplicated code in distgen_internal.cpp
+	unsigned int number_of_cpus = RMSK_SIZE(_syspage_ptr->num_cpu);
+	const int size = number_of_cpus * sizeof(unsigned int) * 2;
+	
+	unsigned int *rsizep = (unsigned int*)malloc(size);
+	if (rsizep == nullptr) {
+		assert(false);
+	}
+	memset((void*)rsizep, 0x00, size);
 
+	*rsizep = number_of_cpus;
+
+	/* Set the runmask. Call this macro once for each processor
+	the thread can run on. */
+	unsigned int *rmaskp = rsizep + 1;
+	RMSK_SET(thread_args->tid, rmaskp);
+
+	/* Set the inherit mask. Call this macro once for each
+	processor the thread's children can run on. */
+	unsigned int *imaskp = rmaskp + number_of_cpus;
+	RMSK_SET(thread_args->tid, imaskp);
+
+	if ( ThreadCtl( _NTO_TCTL_RUNMASK_GET_AND_SET_INHERIT, rsizep) == -1) {
```

Comment:
See comment above.

Reply:
Because this would remove the function call if compiled without asserts.